crypto/string.go

```diff
@@ -0,0 +1,44 @@
+// Copyright 2022 Northern.tech AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+package crypto
+
+import (
+	"go.mongodb.org/mongo-driver/bson"
+)
+
+type String string
+
+const bsonKey = "aes-256"
+
+func (s *String) MarshalBSON() ([]byte, error) {
+	out, err := AESEncrypt(string(*s))
+	if err != nil {
+		return nil, err
+	}
+	return bson.Marshal(map[string][]byte{bsonKey: out})
+}
```

Comment:
Should we store additional metadata? Such as a key identifier so that we can perform seamless key rotations and integrate with KMS at some point?

Reply:
I don't think we need a key identifier, nor integrate with KMS.
Can you please elaborate on why we'd need these things, and how do you envision the (potential) integration with KMS (or Vault, for example)?

Reply:
@alfrunes I updated the implementation to store the name of the algorithm used, together with the binary data. I don't think we need a key ID. The binary data now also contains a SHA-512 signature to verify the encrypted data.

Reply:
> I don't think we need a key identifier, nor integrate with KMS. Can you please elaborate on why we'd need these things, and how do you envision the (potential) integration with KMS (or Vault, for example)?

I was imagining if we have a unique key identifier (e.g. a UUID), there's no need for a "fallback key" since the service will be able to query the key from the KMS if it does not already know about it. This way, key rotations become a simple matter of updating the key id in use, and we'll will still be able to decrypt using previous keys if something should happen during the rotation.
If this pattern sounds ridiculous please ignore this comment :)

Reply:
Two observations:
- we are trying (as much as we can) to not depend on cloud-specific features, and KMS is AWS-only
- the purpose of the key rotation is re-encrypting the data with the new key, thus you still need to read-decrypt-encrypt-write every single value when performing it. I'm not 100% sure I follow your reasoning... what about discussing this topic in the backend team meeting?

Reply:
Ok, let's do a brain-storm some time. BTW I'm completely fine using the fallback-key pattern we use elsewhere, but it would be nice to integrate with a KMS (or CKMS... or whatever the heck the abbreviation for "Secret management services" is :sweat_smile:), which we can reuse for other secrets as well.

Reply:
I'm adding this item to the backend team meeting.